### PR TITLE
Add custom driver to delegate and expose validation

### PR DESCRIPTION
**Pull request by teki**
Sparkle has a nice driver functionality to customize update
machinisms, but it is not exposed. The usage of those
drivers is mostly hard wired. Added the possibility
to supply a custom driver.

I was able to implement most of what we need with
public interfaces, but I had to expose validation
on SUBasicUpdateDriver.

### CODE REVIEW

**Comment by teki** [comment]
@DarrenFordQSR @rusminsusanto @lurkerbot

**Comment by teki** [comment]
Update: exported required symbols and updated the commit message

**Comment by teki** [comment]
@dominic-zhang

**Comment by dominic-zhang** [comment]
:+1:

File: Sparkle/SUUpdater.m [hunk 1]
```diff
@@ -303,7 +303,16 @@ - (void)checkForUpdatesInBackground
     // Background update checks should only happen if we have a network connection.
     //	Wouldn't want to annoy users on dial-up by establishing a connection every
     //	hour or so:
-    SUUpdateDriver *theUpdateDriver = [[([self automaticallyDownloadsUpdates] ? [SUAutomaticUpdateDriver class] : [SUScheduledUpdateDriver class])alloc] initWithUpdater:self];
+    SUUpdateDriver *theUpdateDriver;
```
Comment (DarrenFordQSR) [inline Sparkle/SUUpdater.m]: Might be good to set this to nil (as we check for it later).

I know Apple says [here](https://developer.apple.com/library/ios/releasenotes/ObjectiveC/RN-TransitioningToARC/Introduction/Introduction.html#//apple_ref/doc/uid/TP40011226-CH1-SW5) that it explicitly sets autoreleasing stack variables to nil in ARC, but we know how Apple can flip-flop :-)

I wouldn't worry about this change unless you need to do another build.

**Comment by DarrenFordQSR** [comment]
:+1: :+1: 

If you're gonna do another build then the change might be good (makes it explicit) however don't worry about it if there are no additional changes required.

File: Sparkle/SUUpdater.m (hunk 1 above)
Comment (teki) [inline Sparkle/SUUpdater.m]: done

**Pull request merged**